src/consensus/consensus.h

```diff
@@ -8,17 +8,59 @@
 
 #include <stdint.h>
 
+/** BIP102 block size increase height */
+static const unsigned int BIP102_FORK_MIN_HEIGHT = 485218;
+
 /** The maximum allowed size for a serialized block, in bytes (only for buffer size limits) */
 static const unsigned int MAX_BLOCK_SERIALIZED_SIZE = 4000000;
 /** The maximum allowed weight for a block, see BIP 141 (network rule) */
 static const unsigned int MAX_BLOCK_WEIGHT = 4000000;
 /** The maximum allowed size for a block excluding witness data, in bytes (network rule) */
-static const unsigned int MAX_BLOCK_BASE_SIZE = 1000000;
+static const unsigned int MAX_LEGACY_BLOCK_SIZE = (1 * 1000 * 1000);
+inline unsigned int MaxBlockBaseSize(int nHeight, bool fSegWitActive)
+{
+    if (!fSegWitActive)
+        return MAX_LEGACY_BLOCK_SIZE;
+
+    if (nHeight < (int)BIP102_FORK_MIN_HEIGHT)
+        return MAX_LEGACY_BLOCK_SIZE;
+
+    return (2 * 1000 * 1000);
+}
+
+inline unsigned int MaxBlockBaseSize()
+{
+    return MaxBlockBaseSize(99999999, true);
+}
+
+
 /** The maximum allowed number of signature check operations in a block (network rule) */
-static const int64_t MAX_BLOCK_SIGOPS_COST = 80000;
+static const uint64_t MAX_BLOCK_BASE_SIGOPS = 20000;
+inline int64_t MaxBlockSigOpsCost(int nHeight, bool fSegWitActive)
+{
+    if (!fSegWitActive)
+        return (MAX_BLOCK_BASE_SIGOPS * 4 /* WITNESS_SCALE_FACTOR */);
```

Comment:
The current code base already uses WEIGHT instead of SIZE to verify the block size. It is effectively a MAX_SIZE=1mb as long as there is no witness data.